### PR TITLE
[WIP] move rendering implementations into ITerminal implementations

### PR DESCRIPTION
**Pull request by jonsequitur**

### CODE REVIEW

File: src/System.CommandLine.Rendering/ConsoleExtensions.cs [hunk 1]
```diff
@@ -26,5 +26,31 @@ public static OutputMode DetectOutputMode(this IConsole console)
                 return OutputMode.File;
             }
         }
+
+        public static ITerminal TryGetTerminal(
```
Comment (Keboo) [inline src/System.CommandLine.Rendering/ConsoleExtensions.cs]: Given that this doesn't follow the normal Try* signature pattern, I would suggest simply calling it `GetTerminal`

File: src/System.CommandLine.Rendering/ConsoleExtensions.cs [hunk 2]
```diff
@@ -26,5 +26,31 @@ public static OutputMode DetectOutputMode(this IConsole console)
...
+            }
+            else if (preferVirtualTerminal &&
+                     VirtualTerminalMode.TryEnable() is VirtualTerminalMode virtualTerminalMode &&
+                     virtualTerminalMode.IsEnabled)
+            {
+                return new VirtualTerminal(
+                    console,
+                    virtualTerminalMode);
+            }
+            else
```
Comment (Keboo) [inline src/System.CommandLine.Rendering/ConsoleExtensions.cs]: `else`s not needed in this method

File: src/System.CommandLine.Rendering/AnsiRenderingSpanVisitor.cs [hunk 3]
```diff
@@ -9,8 +9,8 @@ namespace System.CommandLine.Rendering
     internal class AnsiRenderingSpanVisitor : ContentRenderingSpanVisitor
     {
         public AnsiRenderingSpanVisitor(
-            IConsole console,
-            Region region) : base(console.Out, region)
+            IConsole terminal,
```
Comment (Keboo) [inline src/System.CommandLine.Rendering/AnsiRenderingSpanVisitor.cs]: Not sure this rename makes sense since the type is still `IConsole`

**Comment by jonsequitur** [comment]
@dotnet test this please

**Pull request merged**